cmd/internal/cli/apptainer.go

```diff
@@ -105,6 +105,7 @@ var singSilentFlag = cmdline.Flag{
 	Name:         "silent",
 	ShortHand:    "s",
 	Usage:        "only print errors",
+	EnvKeys:      []string{"SILENT"},
```

Comment:
While we are here, can you also add the environment variable `APPTAINER_NOCOLOR` for `--nocolor` flag please ?

Reply:
thanks! added in this commit
https://github.com/apptainer/apptainer/pull/1472/commits/6b6e363dc204933af6d39ec13a7a6d6f33543e40

Have to call `SetLevel` separately, because the `color` will always be `false` in this case, as `!term.IsTerminal(2)` is always true.

Reply:
I don't see any NOCOLOR EnvKeys under the `--nocolor` option, so I don't know what you mean.  When you do add it, please include it in the CHANGELOG.

Also, `!term.IsTerminal(2)` can not always be true.  It has to depend on whether or not stderr is redirected to a file.

Reply:
Oh, maybe you meant in the dummy function it is always redirected to a file.

Reply:
yes, dummy function does not have full implementations for the log writer.

Reply:
thanks for the comments.
Added in this commit
https://github.com/apptainer/apptainer/pull/1472/commits/55907aa873fb1690512bb3f1f9efb16c1ba30e5f